windowsforms/FlexGridShowcaseDemo/cs/MainForm.cs

```diff
@@ -0,0 +1,682 @@
+            };
+
+            // Products table
+            var productsDataTable = new DataTable("Products");
+            
+            var productsColumns = productsDataTable.Columns;
+
+            // Add columns
+            productsColumns.Add("Name");
+            productsColumns.Add("Size", typeof(float));
+            productsColumns.Add("Weight", typeof(float));
+            productsColumns.Add("Quantity", typeof(uint));
+            productsColumns.Add("Description");
+
+            var productsRows = productsDataTable.Rows;
+
+            // Add rows
+            productsRows.Add("Gadget", 120f, 900f, 2, descriptions[_rnd.Next(descriptions.Count - 1)]);
+            productsRows.Add("Widget", 20f, 20f, 25, descriptions[_rnd.Next(descriptions.Count - 1)]);
+            productsRows.Add("Doohickey", 74f, 90f, 100, descriptions[_rnd.Next(descriptions.Count - 1)]);
+
+            var tables = _dataSet.Tables;
+            tables.Add(productsDataTable);
+
+            // Data table
+            var dataDataTable = new DataTable("Data");
+
+            var dataColumns = dataDataTable.Columns;
+
+            // Add columns
+            dataColumns.Add("ID", typeof(int));
+            dataColumns.Add("Product");
+            dataColumns.Add("Country", typeof(Countries));
+            dataColumns.Add("Color", typeof(DrawColors));
+            dataColumns.Add("Price", typeof(decimal));
+            dataColumns.Add("Change", typeof(decimal));
+            dataColumns.Add("History", typeof(int[]));
+            dataColumns.Add("Discount", typeof(decimal));
+            dataColumns.Add("Rating", typeof(int));
+            dataColumns.Add("Active", typeof(bool));
+            dataColumns.Add("Date", typeof(DateTime));
+
+            tables.Add(dataDataTable);
+
+            // Creating relation between products and data
+            _dataSet.Relations.Add("Products_Data",
+                    productsColumns["Name"], dataColumns["Product"]);
+
+            _flexGrid.DataSource = _dataSet;
+            _flexGrid.DataMember = "Data";
+        }
+
+        #endregion
+
+        #region Init FlexGrid
+
+        static Image LoadImage(string recourceName)
+        {
+            // load the picture
+            Image img = null;
+            try
+            {
+                var resource = "FlexGridShowcaseDemo.Properties.Resources";
+                var assembly = Assembly.GetExecutingAssembly();
+
+                var manager = new System.Resources.ResourceManager(resource, assembly);
+                if (manager != null)
+                {
+                    Bitmap bmp = (Bitmap)manager.GetObject(recourceName, CultureInfo.InvariantCulture);
+                    return bmp;
+                }
+            }
+            catch (Exception)
+            {
+            }
```

Comment:
Not a good practice for two reasons:

1. Handling general type Exception.
2. Doing nothing inside the catch. Essentially, just hiding a problem.

Reply:
It was overprotection agains non-found resources, etc.
I've got rid from try/catch block and replaced it with checks for null in necessary places - 8580365acf75967f85040f52f1013358d5fcf44e.